### PR TITLE
Remove Python 3.8 GitHub Actions Runs

### PR DESCRIPTION
**Pull request by jrmccluskey**
Migrates 3.8 actions to 3.9. New workflows were run in #32329, widespread failures are expected on this PR as actions are being run on 3.8 but that python version is not being installed.  Does not remove 3.8 support from Beam, that will be in a followup PR.

Part of #31192

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
R: @damccorm @tvalentyn

File: .test-infra/jenkins/PythonTestProperties.groovy
Comment (tvalentyn) [inline .test-infra/jenkins/PythonTestProperties.groovy:23]: is this file still being used given that we don't use Jenkins?

File: sdks/python/tox.ini
Comment (tvalentyn) [inline sdks/python/tox.ini:274]: keeping just 39 is also fine here, if 310 is not used anywhere

**Comment by tvalentyn** [comment]
LGTM.

You can trigger precommit suites to run off your py38dep fork. For example, I triggered these runs:

beam_PreCommit_PythonLint: https://github.com/apache/beam/actions/runs/10815932208
beam_PreCommit_Python: https://github.com/apache/beam/actions/runs/10815983260

I think the test result will accurately represent what will happen after this PR is merged.

File: .test-infra/jenkins/PythonTestProperties.groovy
Comment (jrmccluskey) [reply .test-infra/jenkins/PythonTestProperties.groovy:23]: I started to update the jenkins files and stopped once I realized what they were, it's worth doing some cleanup here eventually

**Comment by jrmccluskey** [comment]
> LGTM.
> 
> You can trigger precommit suites to run off your py38dep fork. For example, I triggered these runs:
> 
> beam_PreCommit_PythonLint: https://github.com/apache/beam/actions/runs/10815932208 beam_PreCommit_Python: https://github.com/apache/beam/actions/runs/10815983260
> 
> I think the test result will accurately represent what will happen after this PR is merged.

I did kick off some of those manually but didn't grab the run logs, we were green on the precommits I ran.

**Comment by jrmccluskey** [comment]
Linting failed, unhappy with something in yaml IO. I'm re-running to confirm and then I'll fix if needed. I think the game plan here would be to merge on Monday so there aren't any breakages over the weekend?

**Comment by tvalentyn** [comment]
SGTM

**Comment by jrmccluskey** [comment]
Okay the linting failure is a yaml warning that I had already disabled previously, the syntax for disabling that check may be different but it's an easy fix once we're merged. Just need an approval to move forward now.

**Review by tvalentyn — APPROVED** [review]

**Comment by Abacn** [comment]
this may have caused beam_LoadTests_Python_Combine_Flink_Batch failing. For some reason, I still see `--environment_config=gcr.io/apache-beam-testing/beam-sdk/beam_python3.8_sdk:latest`

**Pull request merged**
